bankaccount.go

```diff
@@ -68,6 +71,8 @@ func (a *BankAccountParams) AppendToAsSourceOrExternalAccount(body *form.Values)
 		if a.Default {
 			body.Add("default_for_currency", strconv.FormatBool(a.Default))
 		}
+	} else if len(a.ID) > 0 {
+		body.Add("bank_account", a.ID)
```

Comment:
Would you mind adding a quick test in `bankaccount_test.go` (there should a few for the params in there already) to double-check this behavior? One of the reasons that we had a few sources-related regressions during the `form` changeover was that their parameters tend to have pretty complicated encoding logic, but didn't really have any good testing around it.

Reply:
Hmmm isn't the test in `token/client_test.go` enough? What do you want me to test?

Reply:
When implementing custom encoders like `AppendToAsSourceOrExternalAccount`, it's just kind of nice to know that we're encoding as we expect. `token/client_test.go` is something, but stripe-mock's validations are nowhere near as strict as the real API's, so we can't guarantee that we didn't make an encoding mistake that was allowed through.

There are some encoding tests already in `bankaccount_test.go`, but I was just thinking something like this:

``` go
func TestBankAccountParams_AppendToAsSourceOrExternalAccount(t *testing.T) {
	// Sets ID
	{
		params := &BankAccountParams{ID: "ba_123"}
		body := &form.Values{}
		params.AppendToAsSourceOrExternalAccount(body)
		t.Logf("body = %+v", body)
		assert.Equal(t, []string{"ba_123"}, body.Get("bank_account"))
	}

...
}
```